File: examples/demo_agent.py
"""Demo agent. Run it sandboxed from the examples/ directory:

    agentbox run -p agentbox.policy -- python demo_agent.py
    agentbox replay -p agentbox.policy -- python demo_agent.py
    agentbox show
"""

import agentbox.client as box

notes = box.read_text("data/notes.txt")          # replayable effect
raw = open("data/notes.txt").read()              # direct IO: guard-observed
stamp = box.now()                                # frozen on replay
echoed = box.run(["echo", "sandboxed subprocess"])

box.write_text(
    "out/summary.txt",
    f"[{stamp}] {len(notes)} bytes of notes\n{echoed['stdout']}",
)
print("wrote out/summary.txt at", stamp)

try:
    open("/etc/passwd").read()
except PermissionError as exc:
    print("blocked as expected:", exc)

File: examples/demo_agent.js
/** Node demo agent. Run it sandboxed from the examples/ directory:
 *
 *    agentbox run    -p agentbox.policy -- node demo_agent.js
 *    agentbox replay -p agentbox.policy -- node demo_agent.js
 */
const fs = require("fs");
const box = globalThis.agentbox;

const notes = box.readText("data/notes.txt");        // replayable effect
const raw = fs.readFileSync("data/notes.txt", "utf8"); // direct IO: guard-observed
const stamp = box.now();                             // frozen on replay
const echoed = box.run(["echo", "sandboxed subprocess"]);

box.writeText("out/summary.txt", `[${stamp}] ${notes.length} bytes of notes\n${echoed.stdout}`);
console.log("wrote out/summary.txt at", stamp);

try {
  fs.readFileSync("/etc/passwd", "utf8");
} catch (err) {
  console.log("blocked as expected:", err.message);
}
